utils/shmem.c

```diff
@@ -0,0 +1,74 @@
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+
+char *get_shm_root()
+{
+#ifdef __ANDROID__
+	const char *tmpdir;
+	static char uftrace_dir[PATH_MAX];
```

Comment:
Please do not use a static buffer this way as it's not thread safe.  As it'd be written only and read only after it, we can do something like this

```c
{
    static char shmem_root[PATH_MAX] = "";

    if (shmem_root[0] == '\0') {
        // ...
    }
    return shmem_root;
}

Reply:
Well, it won't be thread-safe with suggested change either. We could switch to dynamic memory but then it would be ugly and slow...